### PR TITLE
Fix `LG_LOST_LOGGER_DUE_TO_WEAK_REFERENCE` SpotBugs violations

### PR DESCRIPTION
**Pull request by basil**
Fixes the following `LG_LOST_LOGGER_DUE_TO_WEAK_REFERENCE` SpotBugs violation:

```
[ERROR] High: Changes to logger could be lost in hudson.logging.LogRecorderManager.doConfigLogger(String, String) [hudson.logging.LogRecorderManager] At LogRecorderManager.java:[line 181] LG_LOST_LOGGER_DUE_TO_WEAK_REFERENCE
```

The supression in `WebAppMain` was a reference to the root logger, which cannot be a weak reference. I'm not sure why it was originally suppressed: perhaps earlier versions of SpotBugs were not capable of detecting that it was a root logger. But in any event, the current version of SpotBugs does understand that it is a root logger and no longer raises a warning. So the suppression is unnecessary and can be removed.

The violation in `doConfigLogger` pertains to the "Log Levels" page, where the user can enter an arbitrary logger name and configure its level. That page lists the loggers that were added to a log recorder in the "Add new log recorder" or "Configure log recorder" page, and references to those loggers are maintained in `LogRecorder#loggers` (so they are not at risk of being garbage collected), so if a user edits the level of one of those loggers there is no issue. But if the user enters the name of a logger that doesn't exist, then SpotBugs has a point: a new logger will be created, its level will be set, and garbage collection may collect the logger, thereby clearing its level. The user would most likely find this behavior unexpected.

The most practical solution seems to be to restrict users from creating new loggers in the "Log Levels" page and limit that page to editing the levels of existing loggers. That way, the user can't shoot themselves in the foot by creating a logger that will be garbage collected later. By forcing users to associate loggers with a log recorder, we prevent this undesirable outcome from being possible and ensure that user-created loggers are both persisted to disk and pinned in memory to prevent garbage collection. So I have added a check in the "Log Levels" page to ensure that the logger already exists. If it doesn't exist, we raise an informative error pointing the user to where they need to go to add the logger to a log recorder.

I tested the change interactively and by adding a new integration test that fails before the changes to `core/src/main` and passes with them.

### Proposed changelog entries

Ensure that loggers exist before setting their log levels. In rare cases, setting the log level of a logger may have had no effect.

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [x] (If applicable) Jira issue is well described
- [x] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [x] Appropriate autotests or explanation to why this change has no tests
- [x] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@mention

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are correct
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins-ci.org/issues/?filter=12146)).

### CODE REVIEW

**Review by timja — APPROVED** [review]

**Comment by basil** [comment]
This PR is now ready for merge. We will merge it after approximately 24 hours if there is no negative feedback. Please see the [merge process documentation](https://github.com/jenkinsci/jenkins/blob/master/docs/MAINTAINERS.adoc#merge-process) for more information about the merge process. Thanks!

**Pull request merged**
